test/micro/org/openjdk/bench/java/nio/channels/SelectOne.java

```diff
@@ -67,23 +67,24 @@ public void setup() throws IOException {
         clients = new ArrayList<SocketChannel>();
         peers = new ArrayList<SocketChannel>();
 
-        ServerSocketChannel listener = ServerSocketChannel.open();
-        listener.bind(new InetSocketAddress(0));
-        SocketAddress remote = listener.getLocalAddress();
+        try (ServerSocketChannel listener = ServerSocketChannel.open()) {
+            listener.bind(new InetSocketAddress(0));
```

Comment:
Is using the wildcard really necessary here? I am concerned that the test is relying on unspecified behaviour - namely:
    - listener is bound to the wildcard
    - listener.getLocalAddress() is *probably* the wildcard
    - later SocketChannel is connected using that address which is still *probably* the wildcard.
For the sake of stability - should the test preferably use InetAddress.getLoopbackAddress() for binding and connecting?

Reply:
This is micro rather than a jtreg test so only suitable for running in very controlled environments. I can of course change it to use the loopback address. The original benchmark did that, but I had to strip it down for openjdk/jdk.